aus_council_scrapers/data.py

```diff
@@ -0,0 +1,220 @@
+                **super().to_dict(),
+                "kind": "council-meeting",
+                "webpage_url": self.webpage_url,
+                "download_url": self.download_url,
+            }
+
+    @staticmethod
+    def from_dict(d: Dict[str, Any]) -> ScraperResult.Notice:
+        match d.get("kind", None):
+            case None:
+                return ScraperResult.CouncilMeetingNotice(
+                    name=d["name"],
+                    datetime=NoticeDate.FuzzyRaw(d["date"], d["time"]),
+                    webpage_url=d["webpage_url"],
+                    download_url=d["download_url"],
+                    location=NoticeLocation.Raw(d.get("location")),
+                )
+
+            case "council-meeting":
+                return ScraperResult.CouncilMeetingNotice.from_dict(d)
+
+            case "notice":
+                return ScraperResult.CouncilMeetingNotice.from_dict(d)
+
+            case _:
+                raise ValueError("nonexhaustive parser")
+
+class NoticeLocation:
+    class Base(ABC):
+        @abstractmethod
+        def to_dict(self: Self) -> Dict[str, str]:
+            pass
+
+        @property
+        @abstractmethod
+        def location_string(self: Self) -> Optional[str]:
+            pass
+
+    @dataclass
+    class Raw(Base):
+        raw: Optional[str]
+
+        def to_dict(self: Self) -> Dict[str, str]:
+            raise NotImplementedError('yet')
+
+        @property
+        def location_string(self: Self) -> Optional[str]:
+            if not self.raw or self.raw.isspace():
+                return None
+
+            cleaned = self.raw.replace(r"\w", " ").strip().lower()
+
+            # Remove council chambers string from location
+            council_chamber_regex = re.compile(r"^council\s?chambers?,?", re.IGNORECASE)
+            cleaned = council_chamber_regex.sub("", cleaned)
+
+            if cleaned == "":
+                return None
+
+            return " ".join((word.capitalize() for word in cleaned.split()))
+
+    @staticmethod
+    def from_dict(d: Dict[str, Any]) -> NoticeLocation.Base:
+        raise NotImplementedError('yet')
+
+class NoticeDate:
+    class Base(ABC):
+        @property
+        @abstractmethod
+        def time(self) -> Optional[datetime.time]:
+            pass
+
+        @property
+        @abstractmethod
+        def date(self) -> datetime.date:
+            pass
+
+        @abstractmethod
+        def to_dict(self: Self) -> Dict[str, str]:
+            pass
+
+        def has_transpired(self, state: str) -> bool:
+            timezone = pytz.timezone(TIMEZONES_BY_STATE[state.upper()])
+            today = datetime.datetime.now(timezone).date()
+            return self.date < today
+
+    @dataclass
+    class SimpleDate(Base):
+        date: datetime.date
+        time: Optional[datetime.time]
+
+        def to_dict(self: Self) -> Dict[str, str]:
+            raise NotImplementedError('yet')
+
+    @dataclass
+    class FuzzyRaw(Base):
```

Comment:
seeing as there are no example of `SimpleDate` in practice, maybe it just makes sense to make this a single type.